yarn-project/end-to-end/src/e2e_lending_contract.test.ts

```diff
@@ -4,105 +4,136 @@ import {
-  const BASE = 10n ** 9n;
+  let lendingAccount: LendingAccount;
+  let lendingSim: LendingSimulator;
+
+  const waitForSuccess = async (tx: SentTx) => {
+    const receipt = await tx.wait();
+    expect(receipt.status).toBe(TxStatus.MINED);
+    return receipt;
+  };
 
-  const deployContracts = async (owner: AztecAddress) => {
+  const deployContracts = async () => {
     let lendingContract: LendingContract;
     let priceFeedContract: PriceFeedContract;
 
-    let collateralAsset: NativeTokenContract;
-    let stableCoin: NativeTokenContract;
+    let collateralAsset: TokenContract;
+    let stableCoin: TokenContract;
 
     {
       logger(`Deploying price feed contract...`);
-      const tx = PriceFeedContract.deploy(wallet).send();
-      logger(`Tx sent with hash ${await tx.getTxHash()}`);
-      const receipt = await tx.wait();
-      expect(receipt.status).toBe(TxStatus.MINED);
+      const receipt = await waitForSuccess(PriceFeedContract.deploy(wallet).send());
       logger(`Price feed deployed to ${receipt.contractAddress}`);
       priceFeedContract = await PriceFeedContract.at(receipt.contractAddress!, wallet);
     }
 
     {
       logger(`Deploying collateral asset feed contract...`);
-      const tx = NativeTokenContract.deploy(wallet, 10000n, owner).send();
-      logger(`Tx sent with hash ${await tx.getTxHash()}`);
-      const receipt = await tx.wait();
-      expect(receipt.status).toBe(TxStatus.MINED);
+      const receipt = await waitForSuccess(TokenContract.deploy(wallet).send());
       logger(`Collateral asset deployed to ${receipt.contractAddress}`);
-      collateralAsset = await NativeTokenContract.at(receipt.contractAddress!, wallet);
+      collateralAsset = await TokenContract.at(receipt.contractAddress!, wallet);
     }
 
     {
       logger(`Deploying stable coin contract...`);
-      const tx = NativeTokenContract.deploy(wallet, 0n, owner).send();
-      logger(`Tx sent with hash ${await tx.getTxHash()}`);
-      const receipt = await tx.wait();
-      expect(receipt.status).toBe(TxStatus.MINED);
+      const receipt = await waitForSuccess(TokenContract.deploy(wallet).send());
       logger(`Stable coin asset deployed to ${receipt.contractAddress}`);
-      stableCoin = await NativeTokenContract.at(receipt.contractAddress!, wallet);
+      stableCoin = await TokenContract.at(receipt.contractAddress!, wallet);
     }
 
     {
       logger(`Deploying L2 public contract...`);
-      const tx = LendingContract.deploy(wallet).send();
-      logger(`Tx sent with hash ${await tx.getTxHash()}`);
-      const receipt = await tx.wait();
-      expect(receipt.status).toBe(TxStatus.MINED);
+      const receipt = await waitForSuccess(LendingContract.deploy(wallet).send());
       logger(`CDP deployed at ${receipt.contractAddress}`);
       lendingContract = await LendingContract.at(receipt.contractAddress!, wallet);
     }
+
+    await waitForSuccess(collateralAsset.methods._initialize(accounts[0]).send());
```

Comment:
awesome pattern, (could also call send in the function to clean it up even more! although send here is nice an explicit)